File: solutions/python/transpose/1/transpose.py
def transpose(text):
    text = text.replace(" ", "~")
    i = 0
    rows = text.split("\n")
    sentences = []
    while True:
        sentence = ""
        flag = True
        for row in rows:
            if i < len(row):
                sentence += row[i]
                is_end = flag = False
            else:
                sentence += " "
        if flag:
            break
        sentences.append(sentence.rstrip())
        i += 1
    return "\n".join(sentences).replace("~", " ")

File: solutions/javascript/transpose/1/transpose.js
//
// This is only a SKELETON file for the 'Transpose' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const transpose = (sentences) => {
  sentences = sentences.map(sentence => sentence.replace(/\s/g, '~'));
  const maxLength = Math.max(...sentences.map(sentence => sentence.length)), result = [];
  for(let i = 0; i < maxLength; i++) {
    let newSentence = '';
    for(const sentence of sentences) {
      newSentence += sentence[i] ?? ' ';
    }
    result.push(newSentence.trimEnd().replace(/~/g, ' '));
  }
  return result;
};
